storm/src/main/storm/mesos/util/MesosCommon.java

```diff
@@ -134,14 +136,34 @@ public static Map getFullTopologyConfig(Map conf, TopologyDetails info) {
 
   public static double topologyWorkerCpu(Map conf, TopologyDetails info) {
     Map topologyConfig = getFullTopologyConfig(conf, info);
-    return Optional.fromNullable((Number) topologyConfig.get(WORKER_CPU_CONF))
-        .or(DEFAULT_WORKER_CPU).doubleValue();
+    Object cpuObject = topologyConfig.get(WORKER_CPU_CONF);
+    double topologyWorkerCpu = DEFAULT_WORKER_CPU;
+    if (cpuObject != null && !(cpuObject instanceof Number)) {
+      LOG.warn("Topology {} has invalid mesos cpu configuration: {} has type {}, when it should be a Number", info.getId(), cpuObject, cpuObject.getClass());
+    } else if (cpuObject != null && (cpuObject instanceof Number)) {
+      topologyWorkerCpu = ((Number) cpuObject).doubleValue();
+      if (topologyWorkerCpu < MESOS_MIN_CPU) {
+        LOG.warn("Topology {} has invalid mesos cpu configuration: {} is below {} which is the mesos minimum", info.getId(), topologyWorkerCpu, MESOS_MIN_CPU);
+        topologyWorkerCpu = DEFAULT_WORKER_CPU;
+      }
```

Comment:
nit: I'd add "setting to DEFAULT_WORKER_CPU", so the user will have a little more information.